joint_trajectory_controller/test/test_trajectory_controller.cpp

```diff
@@ -496,18 +530,36 @@ TEST_P(TrajectoryControllerTestParameterized, test_partial_joint_list)
   updateController(rclcpp::Duration::from_seconds(0.25));
 
   double threshold = 0.001;
-  EXPECT_NEAR(traj_msg.points[0].positions[1], joint_pos_[0], threshold);
-  EXPECT_NEAR(traj_msg.points[0].positions[0], joint_pos_[1], threshold);
-  EXPECT_NEAR(initial_joint3_cmd, joint_pos_[2], threshold)
-    << "Joint 3 command should be current position";
+
+  if (traj_controller_->get_has_position_command_interface())
+  {
+    EXPECT_NEAR(traj_msg.points[0].positions[1], joint_pos_[0], threshold);
+    EXPECT_NEAR(traj_msg.points[0].positions[0], joint_pos_[1], threshold);
+    EXPECT_NEAR(initial_joint3_cmd, joint_pos_[2], threshold)
+      << "Joint 3 command should be current position";
+  }
 
   if (
     std::find(command_interface_types_.begin(), command_interface_types_.end(), "velocity") !=
     command_interface_types_.end())
   {
-    // TODO(anyone): need help here - we should at least estimate the sign of the velocity
-    //     EXPECT_NEAR(traj_msg.points[0].velocities[1], joint_vel_[0], threshold);
-    //     EXPECT_NEAR(traj_msg.points[0].velocities[0], joint_vel_[1], threshold);
+    // estimate the sign of the velocity
+    if (0 > copysign(2.0, traj_msg.points[0].positions[0] - initial_joint2_cmd))
```

Comment:
```suggestion
    // joint rotates forward
    if (traj_msg.points[0].positions[0] - initial_joint2_cmd < 0.0)
```

and please apply a similar change to the other 3 ifs/else ifs below